src/halmos/sevm.py

```diff
@@ -1352,21 +1380,44 @@ def call(
         if not ret_size >= 0:
             raise ValueError(ret_size)
 
+        if arg_size > 0:
+            arg = wload(ex.st.memory, arg_loc, arg_size)
+        else:
+            arg = None
+
         caller = ex.prank.lookup(ex.this, to)
 
-        orig_code = ex.code.copy()
-        orig_storage = deepcopy(ex.storage)
-        orig_balance = ex.balance
-        orig_log = ex.log.copy()
+        def send_callvalue(condition=None) -> None:
+            if (
+                # no balance update for CALLCODE which transfers to itself
+                op == EVM.CALL
+                and not (is_bv_value(fund) and fund.as_long() == 0)
+            ):
+                # assume balance is enough; otherwise ignore this path
+                ex.solver.add(UGE(ex.balance_of(caller), fund))
 
-        if op == EVM.CALL and not (is_bv_value(fund) and fund.as_long() == 0):
-            # no balance update for CALLCODE which transfers to itself
-            ex.balance_update(
-                caller, self.arith(ex, EVM.SUB, ex.balance_of(caller), fund)
-            )
-            ex.balance_update(to, self.arith(ex, EVM.ADD, ex.balance_of(to), fund))
+                # conditional send
+                if condition is not None:
+                    value = If(condition, fund, con(0))
+                else:
+                    value = fund
+
+                ex.balance_update(
+                    caller, self.arith(ex, EVM.SUB, ex.balance_of(caller), value)
+                )
+                ex.balance_update(to, self.arith(ex, EVM.ADD, ex.balance_of(to), value))
```

Comment:
it seems a little odd to dispatch to the ADD/SUB opcode handlers, is it to respect the --smt-add/--smt-sub options?

this would feel cleaner IMO:

```suggestion
                ex.balance_update(caller, ex.balance_of(caller) - value)
                ex.balance_update(to, ex.balance_of(to) + value)
```

Reply:
yes, it is to handle the options.